features/posix_spawn.c

```diff
@@ -36,47 +41,46 @@ int main(int argc, char **argv) {
     posix_spawnattr_t attr;
     int n;
     int status;
-    char *cmd[4];
+    char *cmd[3];
     char tmp[1024];
-    if (argv[2]) {
+    if (argc > 1) {
         _exit(signal(SIGHUP, SIG_DFL) != SIG_IGN);
     }
     signal(SIGHUP, SIG_IGN);
     if (posix_spawnattr_init(&attr)) {
         printf("posix_spawnattr_init() FAILED\n");
-        _exit(0);
+        _exit(2);
```

Comment:
I like that you've added distinct exit statuses to help understand why the feature test failed.